waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -248,13 +248,14 @@
 
 (defn pod->ServiceInstance
   "Convert a Kubernetes Pod JSON response into a Waiter Service Instance record."
-  [{:keys [api-server-url] :as scheduler} pod]
+  [{:keys [api-server-url proxy-options] :as scheduler} pod]
   (try
     (let [;; waiter-app is the first container we register
           restart-count (get-in pod [:status :containerStatuses 0 :restartCount] 0)
           instance-id (pod->instance-id pod restart-count)
           node-name (get-in pod [:spec :nodeName])
-          port0 (get-in pod [:spec :containers 0 :ports 0 :containerPort])
+          port0 (or (some-> (get-in pod [:metadata :annotations :waiter/base-port]) (Integer/parseInt))
```

Comment:
Could we rename this annotation to `waiter/service-port` to reflect that this is the port on which we want Waiter to communicate with this service?

Reply:
Done